File: Examples/Communication/CSharp/Src/Server/Services/TestDuplication.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Firefly;
using Firefly.TextEncoding;
using Communication;
using CommunicationDuplication;
using BaseSystem;

namespace Server.Services
{
    public partial class ServerImplementation
    {
        public event Action<CommunicationDuplication.ErrorEvent> CommunicationDuplicationDotError;

        public CommunicationDuplication.ServerTimeReply CommunicationDuplicationDotServerTime(CommunicationDuplication.ServerTimeRequest r)
        {
            return CommunicationDuplication.ServerTimeReply.CreateSuccess(DateTime.UtcNow.DateTimeUtcToString());
        }
    }
}

File: Examples/Communication/CPP/Src/Server/Services/TestDuplication.cpp
#include "Services/ServerImplementation.h"

#include "BaseSystem/Times.h"
#include "BaseSystem/StringUtilities.h"

#include <memory>
#include <string>

using namespace Communication;
using namespace Server::Services;

/// <summary>服务器时间</summary>
std::shared_ptr<CommunicationDuplication::ServerTimeReply> ServerImplementation::CommunicationDuplicationDotServerTime(std::shared_ptr<CommunicationDuplication::ServerTimeRequest> r)
{
    auto s = wideCharToUtf16(DateTimeUtcToString(UtcNow()));
    return CommunicationDuplication::ServerTimeReply::CreateSuccess(s);
}
